contrib/python/scripts/extract_local_velocity.py

```diff
@@ -0,0 +1,314 @@
+        # based on the refinement_level of the global models.
+        if boundary_name == "north":
+            cut_slice.LowerThreshold = np.max(latitude_bounds) - (180 / 2**(refinement_level + 1))
+            cut_slice.UpperThreshold = np.max(latitude_bounds) + (180 / 2**(refinement_level + 1))
+            
+        if boundary_name == "south":
+            cut_slice.LowerThreshold = np.min(latitude_bounds) - (180 / 2**(refinement_level + 1))
+            cut_slice.UpperThreshold = np.min(latitude_bounds) + (180 / 2**(refinement_level + 1))
+            
+        cut_slice.UpdatePipeline()
+
+    else:
+        raise Exception("Unknown boundary name: " + boundary_name)
+        
+    # Create an array for the radius of the regional slice
+    radius_array = np.arange(min(radius_bounds), max(radius_bounds) + output_radius_resolution, output_radius_resolution)
+
+    # Create arrays for the latitude and longitude for the regional slices depending on which boundary is 
+    # currently within the loop
+    if boundary_name == "west":
+        latitude_array = np.arange(min(latitude_bounds), max(latitude_bounds) + output_lateral_resolution, output_lateral_resolution)
+        longitude_array = np.array([np.min(longitude_bounds)])
+        
+    elif boundary_name == "east":
+        latitude_array = np.arange(min(latitude_bounds), max(latitude_bounds) + output_lateral_resolution, output_lateral_resolution)
+        longitude_array = np.array([np.max(longitude_bounds)])
+    
+    elif boundary_name == "north":
+        latitude_array = np.array([np.max(latitude_bounds)])
+        longitude_array = np.arange(min(longitude_bounds), max(longitude_bounds) + output_lateral_resolution, output_lateral_resolution)
+        
+    elif boundary_name == "south":
+        latitude_array = np.array([np.min(latitude_bounds)])
+        longitude_array = np.arange(min(longitude_bounds), max(longitude_bounds) + output_lateral_resolution, output_lateral_resolution)
+
+    # Calculate the cartesian coordinates on the regional slice
+    cartesian_slice_coords = spherical_to_cartesian(radius_array, np.flip(latitude_array), longitude_array, for_slice=True)
+
+    # Save Cartesian coordinates to .csv file for use later
+    np.savetxt(temp_datafile, X=cartesian_slice_coords, delimiter=',', header="x,y,z", comments='')
+    
+    # This opens the .csv as a paraview object
+    reader = simple.OpenDataFile(temp_datafile)
+    reader.UpdatePipeline()
+
+    # Create a TableToPoints filter
+    tabletopoints = servermanager.filters.TableToPoints()
+    
+    # Assign the TableToPoints object the points of the .csv file
+    tabletopoints.XColumn = 'x'
+    tabletopoints.YColumn = 'y'
+    tabletopoints.ZColumn = 'z'
+    tabletopoints.Input = reader
+    tabletopoints.UpdatePipeline()
+
+    # Take the cut_slice object and interpolate the fields onto the TableToPoints filter above.
```

Comment:
Are there arguments within `ResampleWithDataset` that allow specifying the interpolation method?